libzmq/src/error.rs

```diff
@@ -49,34 +40,36 @@ use std::{
 /// ```
 ///
 /// [`ErrorKind`]: enum.ErrorKind.html
-#[derive(Debug)]
-pub struct Error<T = ()> {
-    inner: Context<ErrorKind>,
+#[derive(Debug, Error)]
+#[error("{inner}")]
+pub struct Error<T: Debug = ()> {
+    #[source]
+    inner: ErrorKind,
     content: Option<T>,
```

Comment:
Maybe this would be better in another PR but I think the error type should rather be:

```
pub struct Error<T = ()> {
    kind: ErrorKind,
    content: T
}
```

...and the `cast` method should be replaced by a `map` method akin to `Option::map`, so that `err.cast()` is replaced by `err.map(|_| ())`.

Reply:
Makes sense, `get()` would probably be:
```rust
fn get(&self) -> &T
```

What about `take()`?
```rust
fn take(self) -> T
```
or just
```rust
impl<T> Into<T> for Error<T>
```

Reply:
or maybe:
```rust
fn take(self) -> (Error<()>, T)
```

Reply:
I think a `fn into_inner(self) -> T` might be more obvious than a `Into<T>` impl.